bpo-34788: Add support for scoped IPv6 addresses (#13772)
python/cpython: Doc/whatsnew/3.9.rst

import attempts.
(Contributed by Ngalim Siregar in :issue:`37444`.)

ipaddress
---------

:mod:`ipaddress` now supports IPv6 Scoped Addresses (IPv6 address with suffix ``%<scope_id>``).

> Scoped IPv6 could be parsed to :class:`ipaddress.IPv6Address` constructor.

Reviewer: I would recommend making some minor grammar changes: 

1) "could be" should instead be "can be" 
2) "parsed to" should be "parsed using"
3) I don't think it's necessary to specify "constructor", that's typically implied when referring to a specific class. 

```suggestion
Scoped IPv6 addresses can be parsed using :class:`ipaddress.IPv6Address`.
```

> If present, scope zone ID is avalaible through :attr:`scope_id` attribute.

Reviewer: 1) Spelling fix: "avalaible" -> "available"
2) Added a "the"

```suggestion
If present, scope zone ID is available through the :attr:`scope_id` attribute.
```

Author: @aeros  Thank you! After checks pass, I`m going to ask for core review via python-dev@python.org again.

> If present, scope zone ID is available through the :attr:`scope_id` attribute.

Reviewer 2: ```suggestion
If present, scope zone ID is available through the :attr:`~ipaddress.IPv6Address.scope_id` attribute.
```
`:attr:` should mention the full name if not used inside a class context itself.

> 

Reviewer: Minor nit: could you remove the empty line of whitespace between the news entry and the "contributed by"?

```suggestion
```

Author: Thank you!

Build and C API Changes
=======================

* Provide :c:func:`Py_EnterRecursiveCall` and :c:func:`Py_LeaveRecursiveCall`
  as regular functions for the limited API. Previously, there were defined as
  macros, but these macros didn't work with the limited API which cannot access
  ``PyThreadState.recursion_depth`` field. Remove ``_Py_CheckRecursionLimit``
  from the stable ABI.
  (Contributed by Victor Stinner in :issue:`38644`.)

> 

Reviewer 2: Please drop these lines, they are added by accident.
Victor decided to move this text to another place.

Author: Thanks a lot! Fixed and also corrected typo in my own surname (what a shame).
